HW #6

olgakuzmins opened the pull request:
1. Added custom ArrayList that allow a user:
     * to Put,
     * Get by index,
     * Delete by index
     * and automatically resize
2. Added custom HashSet that allow a user:
     * to Put,
     * perform "Contains" check,
     * iterate,
     * Delete,
     * automatically resize
     * and keep objects uniqueness.

olgakuzmins committed:
Homework No.6
- added custom ArrayList

olgakuzmins committed:
Homework No.6
- added custom HashSet

AtticSiol commented on src/storage/CustomArrayList.java:
```diff
@@ -0,0 +1,163 @@
+    public T getByIndex(int index) {
+        if (index >= 0 && index < elementsAmount) {
+            return (T) array[index];
+        } else
```
missed brackets

olgakuzmins replied:
fixed

AtticSiol commented on src/storage/CustomArrayList.java:
```diff
@@ -0,0 +1,163 @@
+
+    /*  TO DELETE BY INDEX */
+
+    public void deleteByIndex(int index) {
```
if you check `if (index >= 0 && index < elementsAmount` in `getByIndex`, then it is good approach to check it here as well

olgakuzmins replied:
fixed

AtticSiol commented on src/storage/MainCustomArrayList.java:
```diff
@@ -0,0 +1,30 @@
+package storage;
+
+public class MainCustomArrayList {
+    public static void main(String[] args) {
```
`main(...)` method into the application can be just 1

olgakuzmins replied:
I decided to delete both of the "Main..." files to resolve this problem because I used them to check how CustomHashSet and CustomArrayList work, but I will be able to replace them by unittests only after switching the project to Maven.

AtticSiol replied:
Add example of how to use your collections just in your main method in TicketService, at this moment it will be enough)

olgakuzmins replied:
Added)

olgakuzmins committed:
Homework No.6: made changes after the AtticSiol review dated 15.10.24
- added brackets in getByIndex if condition (typo)
- added if condition for index in deleteByIndex
- deleted MainCustomArrayList and MainCustomHashSet

olgakuzmins committed:
HW-7. Builded project on Maven

olgakuzmins committed:
Merge pull request #6 from olgakuzmins/HW-7-relocationToMaven

HW-7. Builded project on Maven

olgakuzmins committed:
HW-6. Added examples of how to use custom collections in main method in TicketService (review from 17.10.2024)
